gc/base/ParallelDispatcher.cpp

```diff
@@ -387,7 +393,20 @@ MM_ParallelDispatcher::shutDownThreads()
 void
 MM_ParallelDispatcher::wakeUpThreads(uintptr_t count)
 {
-	omrthread_monitor_notify_all(_slaveThreadMutex);	
+	/* This thread should notify and release _slaveThreadMutex asap. Threads waking up will need to
+	 * reacquire the mutex before proceeding with the task.
+	 *
+	 * Hybrid approach to notifying threads:
+	 * 	- Cheaper to do to individual notifies for small set of threads from a large pool
+	 * 	- More expensive to do with individual notifies with large set of threads
+	 */
+	if (count < OMR_MIN((uintptr_t) (_threadCountMaximum / 2), _extensions->dispatcherHybridNotifyThreadBound)) {
```

Comment:
is this (uintptr_t) cast really necessary?

Reply:
Removed it, it's not necessary